File: java/src/main/java/gherkin/ast/Examples.java
package gherkin.ast;

import java.util.Collections;
import java.util.List;

public class Examples extends Node {
    private final List<Tag> tags;
    private final String keyword;
    private final String name;
    private final String description;
    private final TableRow header;
    private final List<TableRow> body;

    public Examples(Location location, List<Tag> tags, String keyword, String name, String description, TableRow header, List<TableRow> body) {
        super(location);
        this.tags = Collections.unmodifiableList(tags);
        this.keyword = keyword;
        this.name = name;
        this.description = description;
        this.header = header;
        this.body = Collections.unmodifiableList(body);
    }

    public String getKeyword() {
        return keyword;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public List<TableRow> getBody() {
        return body;
    }

    public TableRow getHeader() {
        return header;
    }

    public List<Tag> getTags() {
        return tags;
    }
}

File: dotnet/Gherkin/Ast/Examples.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Gherkin.Ast
{
    public class Examples : IHasLocation, IHasDescription, IHasRows, IHasTags
    {
        public IEnumerable<Tag> Tags { get; private set; }
        public Location Location { get; private set; }
        public string Keyword { get; private set; }
        public string Name { get; private set; }
        public string Description { get; private set; }
        public TableRow Header { get; private set; }
        public IEnumerable<TableRow> Body { get; private set; }

        public Examples(Tag[] tags, Location location, string keyword, string name, string description, TableRow header, TableRow[] body)
        {
            Tags = tags;
            Location = location;
            Keyword = keyword;
            Name = name;
            Description = description;
            Header = header;
            Body = body;
        }

        IEnumerable<TableRow> IHasRows.Rows
        {
            get { return new TableRow[] {Header}.Concat(Body); }
        }
    }
}